src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -718,9 +777,17 @@ export const Deposit: React.FC<DepositProps> = ({
         : undefined
 
       const _percentageCryptoAmountPrecisionAfterTxFeesAndSweep =
-        _percentageCryptoAmountPrecisionBeforeTxFees
-          .minus(fromBaseUnit(_estimatedFeesData?.txFeeCryptoBaseUnit ?? 0, asset.precision))
-          .minus(fromBaseUnit(_estimatedSweepFeesData?.txFeeCryptoBaseUnit ?? 0, asset.precision))
+        _percentageCryptoAmountPrecisionBeforeTxFees.gte(
+          bnOrZero(fromBaseUnit(_estimatedFeesData?.txFeeCryptoBaseUnit, asset.precision)).plus(
+            bnOrZero(fromBaseUnit(_estimatedSweepFeesData?.txFeeCryptoBaseUnit, asset.precision)),
+          ),
+        )
+          ? _percentageCryptoAmountPrecisionBeforeTxFees
+              .minus(fromBaseUnit(_estimatedFeesData?.txFeeCryptoBaseUnit ?? 0, asset.precision))
+              .minus(
+                fromBaseUnit(_estimatedSweepFeesData?.txFeeCryptoBaseUnit ?? 0, asset.precision),
+              )
+          : bnOrZero(0)
```

Comment:
preferably-blocking: move the fee-deduction logic for non-UTXOs to sendMax only, to tackle the explicit case of maxes failing because of the missing funds for gas.

Additionally, confirm time looks like it may be missing fees checks for max sends (which is the root cause of users being able to sign and broadcast such Txs, and are still possible if user manually inputs max balance), which will need to be added too.

- This should never run for non-UTXOs, see var name `_percentageCryptoAmountPrecisionAfterTxFeesAndSweep`
- this inherently brings many regression risks to the already flaky logic (I take the blame on that one), namely to UTXO sweep checks
- the flow will now change to fail at input time vs. at confirm time (see https://github.com/shapeshift/web/pull/7606/files?w=1#r1729150184) which doesn't play well with the current input time only estimating fees for the purpose

Reply:
Reverted in [c0cca3f](https://github.com/shapeshift/web/pull/7606/commits/c0cca3f5b1d2711ee907ab0deea35f2cbd0dbe79)